tests.py

```diff
@@ -27,9 +30,14 @@ def run_vsm(self, name, input_data, expected_output, use_initial=True):
 
         process = Popen(cmd, stdin=PIPE, stdout=PIPE)
 
-        output, _ = process.communicate(input=input_data.encode(), timeout=2)
-
-        self.assertEqual(output.decode() , expected_output)
+        output, _ = process.communicate(input=input_data.encode(), timeout=5)
```

Comment:
So the reason why I did it like this is because I can't guarantee that state will be printed correctly. Dictionaries in python (which is what the vars are stored in technically) are stored in random order, so when I print them all it will print them in random order. So passing an expected output will randomly fail if we are expecting state to be in a specific order. I think the way I have it here is the best way of doing it, where you can visually confirm the state for debugging and the tests will fail if are bad rules.